pkg/kubelet/kubelet.go

```diff
@@ -1350,12 +1350,20 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*api.Pod, runningPods []*kubeco
 		return err
 	}
 	errlist := []error{}
-	for i := range found {
-		if !active.Has(string(found[i])) {
-			glog.V(3).Infof("Orphaned pod %q found, removing", found[i])
-			if err := os.RemoveAll(kl.getPodDir(found[i])); err != nil {
-				errlist = append(errlist, err)
-			}
+	for _, uid := range found {
+		if active.Has(string(uid)) {
+			continue
+		}
+		if err := os.Remove(kl.getPodVolumesDir(uid)); err != nil {
```

Comment:
nit: this is a little awkward, the remove can fail for multiple reasons but we seem to be using it as a signal that the volume plugin hasn't unmounted the device. Though it will have the intended effect of retrying till the plugin has.